lib/containers/common.pm

```diff
@@ -105,18 +105,27 @@ sub install_docker_when_needed {
     }
     systemctl('is-active docker');
     systemctl('status docker', timeout => 120);
-    assert_script_run('docker info');
+    record_info('docker', script_output('docker info'));
 }
 
 sub install_buildah_when_needed {
     my $host_os = shift;
-    my @pkgs = qw(buildah);
     if (script_run("which buildah") != 0) {
-        # We may run openSUSE with DISTRI=sle and opensuse doesn't have SUSEConnect
-        activate_containers_module if $host_os =~ 'sles';
-        zypper_call "in @pkgs";
-        record_info('buildah', script_output('buildah info'));
+        if ($host_os eq 'centos') {
+            assert_script_run "dnf -y install buildah", timeout => 120;
+        } elsif ($host_os eq 'ubuntu') {
+            my $version_id = script_output('(. /etc/os-release && echo $VERSION_ID)');
+            my $ubuntu_repo = "https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/xUbuntu_${version_id}";
+            assert_script_run qq(echo "deb $ubuntu_repo/ /" | tee /etc/apt/sources.list.d/devel:kubic:libcontainers:stable.list);
+            assert_script_run "curl -L $ubuntu_repo/Release.key | apt-key add -";
+            assert_script_run "apt-get update", timeout => 900;
+            assert_script_run "apt-get -y install buildah", timeout => 300;
+        } else {
+            activate_containers_module if $host_os =~ 'sles';
+            zypper_call('in buildah', timeout => 900);
```

Comment:
Can you explain why the `timeout` is bigger x3 from ubuntu and x8 in centos? it just looks weird. i guess there is some reason.

Reply:
good question... I copy pasted them from the other calls (docker and centos), but I will put everything to 300 as there is no good reasoning behind...